sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryStrategy.java

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import java.time.Duration;
+
+/**
+ * The interface for determining the retry strategy used in {@link RetryPolicy}.
+ */
+public interface RetryStrategy {
+
+    /**
+     * Max number of retry attempts to be make.
+     * @return The max number of retry attempts.
+     */
+    int getMaxRetries();
```

Comment:
It would be nice to think about having a method `boolean shouldRetry(HttpResponse response)` so that user can control the retry behavior by inspecting the response. If we have such a method then we could remove `getMaxRetries()` method. (since max retry can also be controlled by managing necessary state inside `shouldRetry` implementation)

Reply:
As discussed with @anuchandy, I have added a default implementation for `shouldRetry()` in the interface. Implementations can override if necessary.